### PR TITLE
Allow system to be enabled from management page

### PR DESCRIPTION
**Pull request by Jamiras**
Without having to deploy anything.

Note: this just updates the menu and the `isValidConsole` checks to read from the database instead of a config file.

The download page still requires new emulators to be registered in the `releases.php` file. New console IDs can be added to existing emulators and will be hidden until the console is activated.

SQL to get DB in sync with seed file:
```
UPDATE Console SET manufacturer='Microsoft' where ID=26;
UPDATE Console SET order_column=1 WHERE ID IN (4,12,25,33,8,56,34,54,29,31,64);
UPDATE Console SET order_column=2 WHERE ID IN (6,21,50,11,76,14,30,55,26,59,52);
UPDATE Console SET order_column=3 WHERE ID IN (5,41,51,15,67,22,35);
UPDATE Console SET order_column=4 WHERE ID IN (7,36,1,47);
UPDATE Console SET order_column=5 WHERE ID IN (3,17,9,48);
UPDATE Console SET order_column=6 WHERE ID IN (2,77,10,49);
UPDATE Console SET order_column=7 WHERE ID IN (16,13,39);
UPDATE Console SET order_column=8 WHERE ID IN (19,40);
UPDATE Console SET order_column=9 WHERE ID IN (20,68);
UPDATE Console SET order_column=10 WHERE ID IN (18);
UPDATE Console SET order_column=11 WHERE ID IN (78);
UPDATE Console SET order_column=12 WHERE ID IN (62);
UPDATE Console SET order_column=13 WHERE ID IN (24);
UPDATE Console SET order_column=14 WHERE ID IN (60);
UPDATE Console SET order_column=15 WHERE ID IN (28);
UPDATE Console SET active=1 WHERE ID IN (16,71,72,73,74,75,76,77,78,80,102);
```

### CODE REVIEW

File: app/Filament/Resources/SystemResource/Pages/Edit.php [hunk 1]
```diff
@@ -19,4 +22,13 @@ protected function getHeaderActions(): array
             Actions\RestoreAction::make(),
         ];
     }
+
+    protected function handleRecordUpdate(Model $record, array $data): Model
+    {
+        Cache::forget(CacheKey::SystemMenuList);
```
Comment (Jamiras) [inline app/Filament/Resources/SystemResource/Pages/Edit.php:28]: This forces the menu to get rebuilt after a System is updated.

File: app/Helpers/database/release.php [hunk 2]
```diff
@@ -10,94 +10,20 @@
  * https://github.com/RetroAchievements/rcheevos/blob/develop/src/rcheevos/consoleinfo.c
  * https://github.com/RetroAchievements/rcheevos/blob/develop/test/rcheevos/test_consoleinfo.c
  */
+function getValidConsoleIds(): array
+{
+    return Cache::store('array')->rememberForever('system:active-ids', function () {
+        return System::active()->pluck('ID')->toArray();
+    });
```
Comment (Jamiras) [inline app/Helpers/database/release.php:17]: Keep a list of valid console IDs in each request. Fetching the data to redis wasn't any more efficient than fetching it from the database.

File: app/Helpers/database/release.php [hunk 3]
```diff
@@ -131,14 +57,16 @@ function getReleasesFromFile(): ?array
...
     $emulators = array_filter($releases['emulators'] ?? [], fn ($emulator) => $emulator['active'] ?? false);
-    if ($consoles->isNotEmpty()) {
+    if (!empty($consoles)) {
         return array_map(function ($emulator) use ($consoles) {
             $systems = [];
             foreach ($emulator['systems'] as $system) {
-                $systems[$system] = $consoles[$system];
+                if (array_key_exists($system, $consoles)) {
+                    $systems[$system] = $consoles[$system];
+                }
```
Comment (Jamiras) [inline app/Helpers/database/release.php:69]: This filters out inactive systems from the supports systems list for each emulator on the Downloads page.

File: app/Platform/Services/BeatenGamesLeaderboardService.php [hunk 4]
```diff
@@ -88,7 +88,7 @@ public function buildViewData(Request $request): array|RedirectResponse
         ]);
 
         // Grab all the systems so we can build the system filter options.
-        $allSystems = System::orderBy('Name')->get(['ID', 'Name']);
+        $allSystems = System::gameSystems()->active()->orderBy('Name')->get(['ID', 'Name']);
```
Comment (Jamiras) [inline app/Platform/Services/BeatenGamesLeaderboardService.php:91]: adding gamesystem/active scopes here eliminates the check in the widget.

File: config/systems.php
Comment (Jamiras) [inline config/systems.php:1]: This file is still used to seed the table.

File: resources/views/components/game/multiline-avatar.blade.php [hunk 5]
```diff
@@ -59,10 +73,6 @@ class="w-9 h-9"
                 @if ($consoleName && !$consoleId)
                     <span class="block text-xs tracking-tighter mt-px">{{ $consoleName }}</span>
                 @endif
-
-                @if ($consoleId && !$consoleName)
-                    <span class="block text-xs tracking-tighter mt-px">{{ config('systems')[$consoleId]['name'] }}</span>
-                @endif
```
Comment (Jamiras) [inline resources/views/components/game/multiline-avatar.blade.php:65]: if `consoleId` is set, `consoleName` will be pulled from the model, so the code will always go through the logic on line 54.

File: resources/views/pages-legacy/gameSearch.blade.php [hunk 6]
```diff
@@ -6,14 +6,7 @@
 
 $maxCount = 50;
 
-$consoleList = System::get(['ID', 'Name'])->keyBy('ID')->map(fn ($system) => $system['Name']);
-
-// Remove 'Hubs' and 'Events' from the collection.
-$consolesToRemove = ['Hubs', 'Events'];
-$consoleList = $consoleList->filter(function ($value, $key) use ($consolesToRemove) {
-    return !in_array($value, $consolesToRemove);
-});
```
Comment (Jamiras) [inline resources/views/pages-legacy/gameSearch.blade.php:15]: this is handled by the `gameSystems` scope

File: resources/views/pages-legacy/gameSearch.blade.php [hunk 7]
```diff
@@ -41,11 +34,8 @@
     echo "<p>Show:</p>";
     echo "<select class='w-full sm:w-auto' onchange='window.location = \"/gameSearch.php?o=0&p=$method&i=\" + this.options[this.selectedIndex].value'>";
     foreach ($consoleList as $nextConsoleID => $nextConsoleName) {
-        // 0 is "All Consoles". Don't show consoles that haven't been rolled out yet.
-        if ($nextConsoleID == 0 || isValidConsoleId($nextConsoleID)) {
```
Comment (Jamiras) [inline resources/views/pages-legacy/gameSearch.blade.php:45]: This is handled by the `active` scope. The "0" value is added after executing the query.

**Review by wescopeland — APPROVED** [review]
I've tested activating/deactivating various systems to confirm behavior through the various code changes is functioning correctly.

There's nothing in these changes that leaps out to me as a cause for concern. For the noted redundant queries, we'll inevitably clean that stuff up as the UI continues to evolve, especially with a future move to Inertia+React.

**Pull request merged**